File: Tokens/tokentofriend/contracts/src/test/java/net/corda/samples/tokentofriend/contracts/StateTests.java
package net.corda.samples.tokentofriend.contracts;

import net.corda.samples.tokentofriend.states.CustomTokenState;
import net.corda.testing.node.MockServices;
import org.junit.Test;

public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    @Test
    public void hasMessageFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        CustomTokenState.class.getDeclaredField("message");
        // Is the message field of the correct type?
        assert(CustomTokenState.class.getDeclaredField("message").getType().equals(String.class));
    }
}

File: kotlin-samples/Tokens/tokentofriend/contracts/src/test/kotlin/net/corda/samples/tokentofriend/contracts/StateTests.kt
package net.corda.samples.tokentofriend.contracts

import net.corda.samples.tokentofriend.states.CustomTokenState
import org.junit.Assert
import net.corda.testing.node.MockServices
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()

    @Test
    fun hasMessageFieldOfCorrectType() {
        // Does the message field exist?
        CustomTokenState::class.java.getDeclaredField("message")
        // Is the message field of the correct type?
        Assert.assertEquals(CustomTokenState::class.java.getDeclaredField("message").type, String::class.java)
    }
}